client/scripts/views/components/search_bar.ts

```diff
@@ -225,295 +269,352 @@ const getBalancedContentListing = (unfilteredResults: any[], types: SearchType[]
+
+
+    if(historyList.length > 0) {
+      historyList.push(
+        m(ListItem, {class: 'search-history-no-results upper-border',
+          label: 'Tip: You can use operators like \'single quotes\', and the keyword "or" to limit your search!'})
+      )
+    }
+
+    const activeCommunity = app.community ? app.community.name : vnode.state.searchQuery.communityScope
+    const activeChain = app.activeChainId() || vnode.state.searchQuery.chainScope
+    const scopeTitle = m(ListItem, {class: 'disabled', label: 'Scope'})
+
+    const scopeButtons = [SearchScope.Threads, SearchScope.Replies, SearchScope.Communities, SearchScope.Members]
+      .map(s => {
+        return m(Button, {
+          size: Size.LG,
+          active: vnode.state.searchQuery.searchScope.includes(s),
+          onclick: () => {vnode.state.searchQuery.toggleScope(s)},
+          onmouseover: () => {vnode.state.filterMenuActive = true},
+          onmouseout: () => {vnode.state.filterMenuActive = false},
+          label: s
+        })
+      })
+
+    const filterDropdown =
+      m(List, {
+        class: 'search-results-list',
+      }, [
+        activeCommunity ? [scopeTitle, m(ListItem, {
+          class: 'disabled',
+          label: m(Button, {
+            size: Size.LG,
+            onclick: () => { vnode.state.searchQuery.communityScope =
+              vnode.state.searchQuery.communityScope === activeCommunity
+              ? undefined : activeCommunity },
+            active: vnode.state.searchQuery.communityScope === activeCommunity,
+            onmouseover: () => {vnode.state.filterMenuActive = true},
+            onmouseout: () => {vnode.state.filterMenuActive = false},
+            label: `Search inside community: ${activeCommunity}`
+          }),
+        })] :
+        app.activeChainId() && [scopeTitle, m(ListItem, {
+          class: 'disabled',
+          label: m(Button, {
+            size: Size.LG,
+            onclick: () => { vnode.state.searchQuery.chainScope =
+              vnode.state.searchQuery.chainScope === activeChain
+              ? undefined : activeChain },
+            active: vnode.state.searchQuery.chainScope === activeChain ,
+            onmouseover: () => {vnode.state.filterMenuActive = true},
+            onmouseout: () => {vnode.state.filterMenuActive = false},
+            label: `Search inside chain: ${activeChain}`
+          }),
+        })],
+        m(ListItem, {
+          class: 'disabled',
+          label: "I'm looking for: "
+        }),
+        m(ListItem, {
+          class: 'disabled bottom-border search-filter-button-bar',
+          label: scopeButtons}),
+        vnode.state.searchTerm.length < 1 ?
+          historyList.length === 0
+          ? m(ListItem, {class: 'search-history-no-results',
+                         label: 'Enter a term into the field and press Enter to start' })
+          : [m(ListItem, {class: 'disabled', label: 'Search History' }), historyList]
```

Comment:
I like this "Search History" feature a lot, but right now, it's only indexing when the user hits Enter/goes to the full search results page (instead of preview). Possible to index any result that triggers a db query/gets results fetched for it?

Reply:
The search history feature is only storing queries in the browser localStorage to it can persist between site visits. There's still a search cache which deals with storing the actual results - that cache any db search call, preview or not